src/Common/Repositories/Abstract/IActionPlanRepository.cs

```diff
@@ -4,5 +4,5 @@ namespace Middleware.Common.Repositories;
 
 public interface IActionPlanRepository : IBaseRepository<ActionPlanModel>
 {
-    
+    Task<List<ActionPlanModel>> GetActionPlanModelsAsync(Guid robotId);
```

Comment:
I don't understand what this function does by it's name. I had to go to the code to see what exactly it does. Please change it to  for example `GetActionPlansForRobotAsync` because this is what it does right? :)